src/Sylius/Component/Core/Promotion/Checker/TaxonInPromotionRuleChecker.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Promotion\Checker;
+
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Core\Promotion\Checker\Rule\TotalOfItemsFromTaxonRuleChecker;
+use Sylius\Component\Promotion\Model\PromotionRuleInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+final class TaxonInPromotionRuleChecker implements TaxonInPromotionRuleCheckerInterface
+{
+    public function __construct(private RepositoryInterface $promotionRuleRepository)
+    {
+    }
+
+    public function isInUse(TaxonInterface $taxon): bool
+    {
+        $promotionRules = $this->promotionRuleRepository->findBy(['type' => TotalOfItemsFromTaxonRuleChecker::TYPE]);
```

Comment:
What about `HasTaxonRuleChecker::TYPE`? Removing taxons from this type of rule can lead to the same issue.

Reply:
I tested it, it is an array and it works, even if it's empty it just doesn't apply for any taxon and you can access the edit page also